### PR TITLE
[NUXE-331] Kit search

### PR DESCRIPTION
**Pull request by bh247484**
#### What does this PR do?

1. Creates a search bar for kits
2. Reimagines top navbar to help accomodate kit search (new dark mode toggle, smaller nav kit).
3. Adds new horizontal subtle variant to nav kit.
4. Improves doc site React component file structure and js pipeline.
5. Resolves conflicting duplicate WebpackerReact registrations.
6. Cleans up js packs includes.

**Bonus**

7. Fixes bug where dark mode cookie carries over to Visual Guidelines page.  Moves cookie deleting method into `pb_doc_helper` so it can be used in multiple controllers.

#### Screens

top_nav_search
light_search
dark_search

#### Breaking Changes

Some changes to controller logic/helpers but shouldn't break any existing functionality on site.

#### Runway Ticket URL

[Link to Runway Story](https://nitro.powerhrg.com/runway/backlog_items/NUXE-331)

#### How to test this

Test out new features in Milano deploy.  Make sure mobile sizing works as expected.

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

**Review by jasperfurniss — APPROVED** [review]

File: playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss [hunk 1]
```diff
@@ -26,20 +26,49 @@ body {
...
+      position: relative;
+      bottom: $space_md - 4;
+    }
+    .separator-explicit-height {
+      height: $space_xl + 5;
+      margin-left: $space_xs;
+      margin-right: $space_xs;
+    }
+    .toggle-icon {
+      color: $text_lt_light;
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss:53]: image

Should be lighter. Why not use a body kit do you don't have to set this?

**Review by jasoncypret — CHANGES_REQUESTED** [review]
image

Dark needs to be pass through

File: playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss (hunk 1 above)
Comment (jasoncypret) [reply playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss:53]: Can we also make the icon clickable? Let me know if that would be easy or not

Comment (bh247484) [reply playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss:53]: Good call, didn't think to wrap icon in body kit.

**Comment by bh247484** [comment]
> image
> 
> Dark needs to be pass through

Looks like there is no dark mode version for Toggle kit.

File: playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss (hunk 1 above)
Comment (bh247484) [reply playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss]: Can make the icon clickable (wrap in a tag with href pointed to dark mode method routes) but it won't trigger the toggle on click.  Might confuse users.

Comment (bh247484) [reply playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss]: Actually I'll see if there's a way to trip the toggle with javascript and create an onClick handler for the icon.

Comment (bh247484) [reply playbook/app/pb_kits/playbook/packs/site_styles/_site-style.scss]: Yep that works.

**Pull request merged**